packages/shared/util.ts

```diff
@@ -14,3 +14,30 @@ export function truncateText(text: string, max_len: number) {
 export function isInviteLink(url: string) {
   return url.startsWith('https://i.delta.chat/') && url.includes('#')
 }
+
+export function throttle<R, A extends any[]>(
+  fn: (...args: A) => R,
+  wait: number
+) {
+  let inThrottle: boolean,
+    timeout: ReturnType<typeof setTimeout>,
+    lastTime: number
+  return (...args: A) => {
+    if (!inThrottle) {
+      fn(...args)
+      lastTime = performance.now()
+      inThrottle = true
+    } else {
+      clearTimeout(timeout)
+      timeout = setTimeout(
+        () => {
+          if (performance.now() - lastTime >= wait) {
```

Comment:
So if this condition is false, the last call of the throttled function will not invoke `fn`?

Why not use an existing package?